concent-builder/configure-jenkins.yml

```diff
@@ -0,0 +1,184 @@
+            - concent-secrets/cloud/
+            - concent-secrets/jenkins/
+
+    - become:      yes
+      become_user: root
+      block:
+        - name: Upload secrets
+          copy:
+            src:   "{{ local_secret_dir }}/{{ cluster }}/{{ item }}"
+            dest:  "{{ jenkins_home_dir }}/concent-secrets/{{ cluster }}/{{ item }}"
+            owner: jenkins
+            group: jenkins
+          with_items:
+            - cluster-secrets.yml
+            - "../cloud/{{ gke.service_account_name }}-private-key.json"
+            - nginx-proxy-ssl.crt
+            - nginx-proxy-ssl.key
+            - nginx-storage-ssl.crt
+            - nginx-storage-ssl.key
+            - "../jenkins/.token"
+
+        - name: Delete default user settings
+          file:
+            path:  /var/lib/jenkins/users/users.xml
+            state: absent
+
+        - name: Create directory for jenkins job and workspace
+          file:
+            path:  "{{ item }}"
+            state: directory
+            owner: jenkins
+            group: jenkins
+          with_items:
+            - "/var/lib/jenkins/jobs/Build-deploy-and-test/"
+            - "/var/lib/jenkins/jobs/Build-deploy-and-test/workspace"
+            - "/var/lib/jenkins/users/admin/"
+            - "/var/lib/jenkins/secrets/"
+
+        - name: Upload jenkins secrets
+          copy:
+            src:   "{{ local_secret_dir }}/jenkins/{{ item.name }}"
+            dest:  "/var/lib/jenkins{{ item.path }}{{ item.name }}"
+            owner: "jenkins"
+            group: "jenkins"
+          with_items:
+            - { path: "/users/admin/", name: "config.xml" }
+            - { path: "/users/admin/", name: "apiTokenStats.xml" }
+            - { path: "/", name: "credentials.xml" }
+            - { path: "/secrets/", name: "master.key" }
+            - { path: "/secrets/", name: "hudson.util.Secret" }
+
+        - name: Upload jenkins configuration file
+          # FIXME This need use as remote_src
+          copy:
+            src:   "jenkins-configuration-files/{{ item.name }}"
+            dest:  "/var/lib/jenkins{{ item.path }}{{ item.name }}"
+            owner: "jenkins"
+            group: "jenkins"
+          with_items:
+            - { path: "/jobs/Build-deploy-and-test/", name: "config.xml" }
+            - { path: "/", name: "github-plugin-configuration.xml" }
+            - { path: "/", name: "jenkins.model.JenkinsLocationConfiguration.xml" }
+
+        - name: Restart jenkins service
+          service:
+            enabled: yes
+            name:    jenkins
+            state:   restarted
+
+    - become:      yes
+      become_user: jenkins
+      block:
+        - name: Wait for jenkins after restart daemon
+          wait_for:
+            port:  8080
+            delay: 60
+
+        - name: Download jenkins-cli
+          get_url:
+            url:   "http://{{ jenkins_address }}/jnlpJars/jenkins-cli.jar"
+            dest:  "{{ jenkins_home_dir }}/jenkins-cli.jar"
```

Comment:
Please include Jenkins version in the file name. Otherwise this playbook won't download a newer version when we update Jenkins.

BTW, is the file stored on our server or is it just a redirect to another machine? If it's stored locally you could just get a path to it and run it directly without downloading it. This would solve the problem of updating it - it would get updated automatically with Jenkins.

Reply:
https://wiki.jenkins.io/display/JENKINS/Jenkins+CLI#JenkinsCLI-ObtainingtheCLIclient

Reply:
It does not really say if it's storing it on the sever. I only see that you're using a local URL but the server could just as well be redirecting you to some remote location.

Could you try to find it on disk when Jenkins is installed?

Reply:
I found the `jenkins-cli.jar` file on the local disk. The task now copy this file to home jenkins directory.

Reply:
Please make it a symlink.